### PR TITLE
Clean up six / future / past and other Py2 detritus

### PR DESCRIPTION
**Pull request by mr-c**

### CODE REVIEW

File: src/toil/test/__init__.py [hunk 1]
```diff
@@ -749,7 +744,7 @@ def __init__(self, outer, mounts, cleanMounts=False):
             """
             :param ApplianceTestSupport outer:
             """
-            assert all(' ' not in v for v in itervalues(mounts)), 'No spaces allowed in mounts'
+            assert all(' ' not in v for v in mounts.value()), 'No spaces allowed in mounts'
```
Comment (DailyDreaming) [inline src/toil/test/__init__.py:747]: `.values()`

I'm guessing this section is untested.

File: src/toil/batchSystems/abstractBatchSystem.py [hunk 2]
```diff
@@ -65,7 +66,7 @@ class BatchJobExitReason(enum.Enum):
     'cleanWorkDir'))
 
 
-class AbstractBatchSystem(with_metaclass(ABCMeta, object)):
+class AbstractBatchSystem(object, metaclass=ABCMeta):
```
Comment (DailyDreaming) [inline src/toil/batchSystems/abstractBatchSystem.py:69]: Consider dropping the implied `object`:

```class AbstractBatchSystem(metaclass=ABCMeta):```

or

```class AbstractBatchSystem(ABC):```

File: src/toil/fileStores/abstractFileStore.py [hunk 3]
```diff
@@ -11,25 +11,24 @@
...
 
+from toil.common import cacheDirName
 from toil.fileStores import FileID
+from toil.lib.misc import WriteWatchingStream
+from toil.lib.objects import abstractclassmethod
 
 logger = logging.getLogger(__name__)
 
-class AbstractFileStore(with_metaclass(ABCMeta, object)):
+class AbstractFileStore(object, metaclass=object):
```
Comment (DailyDreaming) [inline src/toil/fileStores/abstractFileStore.py:31]: Consider dropping `object` here.

File: src/toil/jobStores/utils.py [hunk 4]
```diff
@@ -131,7 +127,7 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 os.close(readable_fh)
 
 
-class ReadablePipe(with_metaclass(ABCMeta, object)):
+class ReadablePipe(object, metaclass=ABCMeta):
```
Comment (DailyDreaming) [inline src/toil/jobStores/utils.py:130]: Consider dropping `object` here.

File: src/toil/jobStores/utils.py [hunk 5]
```diff
@@ -1,17 +1,13 @@
...
-from abc import abstractmethod
+from abc import ABCMeta, abstractmethod
 
 from toil.lib.threading import ExceptionalThread
-from future.utils import with_metaclass
 
 log = logging.getLogger(__name__)
 
-class WritablePipe(with_metaclass(ABCMeta, object)):
+class WritablePipe(object, metaclass=ABCMeta):
```
Comment (DailyDreaming) [inline src/toil/jobStores/utils.py:10]: Consider dropping `object` here.

File: src/toil/lib/generatedEC2Lists.py [hunk 6]
```diff
@@ -14,11 +14,9 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from six import iteritems
 from toil.lib.ec2nodes import InstanceType
 
-
-# 271 Instance Types.  Generated 2020-02-07 12:22:28.845267.
+# 342 Instance Types.  Generated 2020-11-16 13:11:17.945619.
```
Comment (DailyDreaming) [inline src/toil/lib/generatedEC2Lists.py:19]: AWS has been busy I see.  >.>

File: src/toil/lib/threading.py [hunk 7]
```diff
@@ -92,9 +89,15 @@ def tryRun( self ):
...
+            exc_type, exc_value, traceback = self.exc_info
             self.exc_info = None
-            raise_(type, value, traceback)
+            if exc_value is not None:
+                exc = exc_type(exc_value)
+            else:
+                exc = exc_type
+            if exc.__traceback__ is not traceback:
+                raise exc.with_traceback(traceback)
+            raise exc
```
Comment (DailyDreaming) [inline src/toil/lib/threading.py:100]: I've seen this incantation twice now (in the `exceptions.py` file too).  Consider making these into one function.

File: src/toil/provisioners/abstractProvisioner.py [hunk 8]
```diff
@@ -98,7 +97,7 @@ def __hash__(self):
              self.preemptable))
 
 
-class AbstractProvisioner(with_metaclass(ABCMeta, object)):
+class AbstractProvisioner(object, metaclass=ABCMeta):
```
Comment (DailyDreaming) [inline src/toil/provisioners/abstractProvisioner.py:100]: Consider dropping `object` here.

File: src/toil/realtimeLogger.py [hunk 9]
```diff
@@ -103,7 +97,7 @@ def __getattr__(self, name):
         return getattr(self.getLogger(), name)
 
 
-class RealtimeLogger(with_metaclass(RealtimeLoggerMetaclass, object)):
+class RealtimeLogger(object, metaclass=RealtimeLoggerMetaclass):
```
Comment (DailyDreaming) [inline src/toil/realtimeLogger.py:100]: Consider dropping `object` here.

File: src/toil/test/__init__.py [hunk 10]
```diff
@@ -29,17 +28,13 @@
...
-from builtins import str
-from future.utils import with_metaclass
-from six import iteritems, itervalues
-from six.moves.urllib.request import urlopen
-from unittest.util import strclass
 
-from toil import subprocess
-from toil import toilPackageDirPath, applianceSelf, ApplianceImageNotFound
-from toil import which
+from toil import (ApplianceImageNotFound, applianceSelf, subprocess,
```
Comment (DailyDreaming) [inline src/toil/test/__init__.py:36]: Both `subprocess` and `which` here were only imported from toil to get py2/3 compatible imports, so they can now be:

```
import subprocess
from shutil import which
```

File: src/toil/test/src/miscTests.py [hunk 11]
```diff
@@ -178,12 +176,18 @@ def try_and_panic_by_hand(self):
...
             except Exception:
                 pass
-            raise_(exc_type, exc_value, exc_traceback)
+            if exc_value is not None:
+                exc = exc_type(exc_value)
+            else:
+                exc = exc_type
+            if exc.__traceback__ is not traceback:
+                raise exc.with_traceback(traceback)
+            raise exc
```
Comment (DailyDreaming) [inline src/toil/test/src/miscTests.py:190]: Same incantation from before.

**Review by DailyDreaming — COMMENTED** [review]
This is awesome, thanks!  Only one comment that needs to be fixed, the rest are nits.  Thanks again for the clean up!

File: src/toil/lib/generatedEC2Lists.py (hunk 6 above)
Comment (mr-c) [reply src/toil/lib/generatedEC2Lists.py:19]: :-D

File: src/toil/lib/threading.py (hunk 7 above)
Comment (mr-c) [reply src/toil/lib/threading.py]: Done, thanks!

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Thank you @mr-c .

**Pull request merged**